cfme/tests/integration/test_cfme_auth.py

```diff
@@ -1,90 +1,156 @@
 import pytest
-from widgetastic_patternfly import CandidateNotFound
+from six import iteritems
 
 from cfme.base.credential import Credential
+from cfme.utils.auth import (
+    OpenLDAPAuthProvider, OpenLDAPSAuthProvider, ActiveDirectoryAuthProvider, FreeIPAAuthProvider,
+    AmazonAuthProvider
+)
 from cfme.utils.blockers import GH
 from cfme.utils.conf import auth_data
 from cfme.utils.log import logger
+from cfme.utils.wait import wait_for
 
-pytestmark = pytest.mark.uncollectif(lambda appliance: appliance.is_pod)
+pytestmark = [
+    pytest.mark.uncollectif(lambda appliance: appliance.is_pod),
+    pytest.mark.meta(blockers=[
+        GH('ManageIQ/integration_tests:6465',
+           # need SSL openldap server
+           unblock=lambda auth_mode, prov_key: not (
+               auth_mode in ['external', 'ldaps'] and
+               auth_data.auth_providers[prov_key].type == 'openldaps')
+           )
+    ])
+]
 
-RETRIEVE_GROUP = 'retrieve_group'
-CREATE_GROUP = 'create_group'
+# map auth provider types, auth_modes, and user_types for test matrix
+# first key level is auth mode
+# second key level is provider type  (auth_provider key in parametrization)
+# finally, user_types valid for testing on the above combination of provider+mode
+test_param_maps = {
+    'amazon': {
+        AmazonAuthProvider.auth_type: {
+            'user_types': ['username']}
+    },
+    'ldap': {
+        ActiveDirectoryAuthProvider.auth_type: {
+            # add cn_domain, samacct
+            'user_types': ['cn', 'email', 'uid', 'upn']
+        },
+        FreeIPAAuthProvider.auth_type: {
+            'user_types': ['cn', 'uid']  # add cn_domain
+        },
+        OpenLDAPAuthProvider.auth_type: {
+            'user_types': ['cn', 'uid']  # add cn_domain
+        }
+    },
+    'external': {
+        FreeIPAAuthProvider.auth_type: {
+            'user_types': ['uid']
+        },
+        OpenLDAPSAuthProvider.auth_type: {
+            'user_types': ['uid']
+        }
+        # TODO add ActiveDirectory SAMAcct usertype for external
+    }}
 
 
 def pytest_generate_tests(metafunc):
-    auth_modes = auth_data.get('auth_providers')
-    if not auth_modes:
-        return
-    argvalues = [[authmode] for authmode in auth_modes]
-    if 'configure_auth' in metafunc.fixturenames:
-        metafunc.parametrize(['auth_mode'], argvalues)
-
-
-@pytest.fixture()
-def data(request, auth_mode, add_group):
-    yaml_data = auth_data['auth_test'].get(auth_mode, {})
-    if add_group == 'evm_default_group':
-        yaml_data['get_groups'] = False
-    return yaml_data
-
-
-@pytest.fixture()
-def group(request, data, auth_mode, add_group, appliance):
-    if not data:
-        pytest.skip("No data spcified for user group")
-    credentials = Credential(
-        principal=data["username"],
-        secret=data["password"],
-    )
+    """ zipper auth_modes and auth_prov together and drop the nonsensical combos """
+    argnames = ['auth_mode', 'prov_key', 'user_type']
```

Comment:
**REQUIRED:** This is OK for now, but will need to be converted to supportability type when provider type/version drops

Reply:
Adding TODO